src/test/java/io/cdap/plugin/splunk/common/config/BaseSplunkConfigTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.splunk.common.config;
+
+import io.cdap.cdap.etl.api.validation.CauseAttributes;
+import io.cdap.cdap.etl.api.validation.ValidationFailure;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+
+/**
+ * Tests for {@link BaseSplunkConfig}
+ */
+public class BaseSplunkConfigTest {
+
+  private static final String MOCK_STAGE = "mockStage";
+
+  @Test
+  public void testValidate() {
+    BaseSplunkConfig config = new BaseSplunkConfig(
+      "reference",
+      "basic",
+      "apiToken",
+      "userName",
+      60000,
+      60000,
+      3,
+      60000,
+      100);
+
+    MockFailureCollector collector = new MockFailureCollector(MOCK_STAGE);
+    config.validate(collector);
+    Assert.assertTrue(collector.getValidationFailures().isEmpty());
+  }
+
+  @Test
+  public void testInvalidAuthenticationType() {
```

Comment:
only 2 tests?

Reply:
BaseSplunkConfig does validation only for one field. All other validations are done in SplunkBatchSourceConfig.